libraries/app/api.cpp

```diff
@@ -316,12 +316,11 @@ namespace graphene { namespace app {
        auto index_start = by_op_idx.begin();
        auto itr = by_op_idx.lower_bound(boost::make_tuple(account, start));
 
-       while(itr != index_start && itr->operation_id.instance.value > stop.instance.value && result.size() < limit)
+       while(itr != index_start && itr->account == account && itr->operation_id.instance.value > stop.instance.value && result.size() < limit)
        {
-          if(itr->account == account && itr->operation_id.instance.value <= start.instance.value)
+          if(itr->operation_id.instance.value <= start.instance.value)
```

Comment:
I think this condition is always true and can be removed (itr starts at lower_bound(account, start).

Reply:
>I think this condition is always true and can be removed (itr starts at lower_bound(account, start).

No. `lower_bound` implies `greater` if the identical one can't not be found, which should not be included in the result.

Reply:
You're right, with max_ops_per_account and partial_operations start might not exist.